app/src/main/java/com/nextcloud/talk/controllers/bottomsheet/EntryMenuController.java

```diff
@@ -143,54 +144,49 @@ protected void onAttach(@NonNull View view) {
     @OnClick(R.id.ok_button)
     public void onProceedButtonClick() {
         Bundle bundle;
-        if (operationCode == 99) {
-            eventBus.post(new BottomSheetLockEvent(false, 0, false, false));
+        if (operation == ConversationOperationEnum.JOIN_ROOM) {
             bundle = new Bundle();
             bundle.putParcelable(BundleKeys.INSTANCE.getKEY_ROOM(), Parcels.wrap(conversation));
             bundle.putString(BundleKeys.INSTANCE.getKEY_CALL_URL(), callUrl);
             bundle.putString(BundleKeys.INSTANCE.getKEY_CONVERSATION_PASSWORD(), editText.getText().toString());
-            bundle.putInt(BundleKeys.INSTANCE.getKEY_OPERATION_CODE(), operationCode);
+            bundle.putSerializable(BundleKeys.INSTANCE.getKEY_OPERATION_CODE(), operation);
             if (originalBundle.containsKey(BundleKeys.INSTANCE.getKEY_SERVER_CAPABILITIES())) {
                 bundle.putParcelable(BundleKeys.INSTANCE.getKEY_SERVER_CAPABILITIES(), originalBundle.getParcelable(BundleKeys.INSTANCE.getKEY_SERVER_CAPABILITIES()));
             }
 
             getRouter().pushController(RouterTransaction.with(new OperationsMenuController(bundle))
                     .pushChangeHandler(new HorizontalChangeHandler())
                     .popChangeHandler(new HorizontalChangeHandler()));
-        } else if (operationCode != 7 && operationCode != 10 && operationCode != 11) {
-            eventBus.post(new BottomSheetLockEvent(false, 0, false, false));
+        } else if (operation != ConversationOperationEnum.SHARE_LINK && operation != ConversationOperationEnum.GET_AND_JOIN_ROOM && operation != ConversationOperationEnum.INVITE_USERS) {
             bundle = new Bundle();
-            if (operationCode == 4 || operationCode == 6) {
+            if (operation == ConversationOperationEnum.CHANGE_PASSWORD || operation == ConversationOperationEnum.SET_PASSWORD) {
```

Comment:
Can you add an on-demand static import for `ConversationOperation` enum?

```
import static com.nextcloud.talk.controllers.bottomsheet.ConversationOperationEnum.*;
```

In my opinion it will be much more readable.

Reply:
done